src/pages/workspace/accounting/intacct/EnterSageIntacctCredentialsPage.tsx

```diff
@@ -0,0 +1,117 @@
+        // here we call ConnectToSageIntacct API command
+        Navigation.goBack();
+    }, []);
+
+    const validate = useCallback((values: FormOnyxValues<typeof ONYXKEYS.FORMS.SAGE_INTACCT_CREDENTIALS_FORM>) => {
+        const errors: FormInputErrors<typeof ONYXKEYS.FORMS.SAGE_INTACCT_CREDENTIALS_FORM> = {};
+
+        if (!values.companyID) {
+            ErrorUtils.addErrorMessage(errors, 'companyID', 'common.error.fieldRequired');
+        }
+
+        if (!values.userID) {
+            ErrorUtils.addErrorMessage(errors, 'userID', 'common.error.fieldRequired');
+        }
+
+        if (!values.password) {
+            ErrorUtils.addErrorMessage(errors, 'password', 'common.error.fieldRequired');
+        }
+        return errors;
+    }, []);
+    return (
+        <ScreenWrapper
+            includeSafeAreaPaddingBottom={false}
+            shouldEnableMaxHeight
+            testID={EnterSageIntacctCredentialsPage.displayName}
+        >
+            <HeaderWithBackButton
+                title={translate('workspace.intacct.sageIntacctSetup')}
+                onBackButtonPress={() => Navigation.goBack()}
+            />
+            <FormProvider
+                style={[styles.flexGrow1, styles.ph5]}
+                formID={ONYXKEYS.FORMS.SAGE_INTACCT_CREDENTIALS_FORM}
+                validate={validate}
+                onSubmit={confirmCredentials}
+                submitButtonText={translate('common.confirm')}
+                enabledWhenOffline
+                shouldValidateOnBlur
+                shouldValidateOnChange
+            >
+                <Text style={[styles.textHeadlineH1, styles.pb5, styles.pt3]}>{translate('workspace.intacct.enterCredentials')}</Text>
+                <View style={styles.mb4}>
+                    <InputWrapper
+                        InputComponent={TextInput}
+                        inputID={INPUT_IDS.COMPANY_ID}
+                        name="companyID"
+                        label={translate('common.companyID')}
+                        aria-label={translate('common.companyID')}
+                        role={CONST.ROLE.PRESENTATION}
+                        defaultValue={sageIntacctCredentialsDraft?.[INPUT_IDS.COMPANY_ID]}
+                        spellCheck={false}
+                    />
+                </View>
+                <View style={styles.mb4}>
+                    <InputWrapper
+                        InputComponent={TextInput}
+                        inputID={INPUT_IDS.USER_ID}
+                        name="userID"
+                        label={translate('common.userID')}
+                        aria-label={translate('common.userID')}
+                        role={CONST.ROLE.PRESENTATION}
+                        defaultValue={sageIntacctCredentialsDraft?.[INPUT_IDS.USER_ID]}
+                        spellCheck={false}
+                    />
+                </View>
```

Comment:
[Nit]: Those (and the next one) could probably be abstracted to a function with some args

Reply:
up to you :) (personally, I would do this but no hard feelings)